src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/TypeBuilder.cs

```diff
@@ -497,23 +485,24 @@ public GCLayout(RuntimeTypeHandle rtth)
             /// <param name="bitfield">The bitfield to write a layout to (may be null, at which
             /// point it will be created and assigned).</param>
             /// <param name="offset">The offset at which we need to write the bitfield.</param>
-            public void WriteToBitfield(LowLevelList<bool> bitfield, int offset)
+            /// <returns>The result bitfield, may or may not be the input array.</returns>
+            public bool[] WriteToBitfield(bool[] bitfield, int offset)
```

Comment:
I'm not sure how relevant is this. I can only find one caller of this method which builds ArrayType.

Reply:
I am not sure what you mean by "this"

Reply:
I mean the `WriteToBitfield` method and its descendants. I can only find one caller and its input argument is always empty. The GCLayout structure represents a mergeable bit vector, but I can't find the callers fully leveraging it.

Reply:
Feel free to simplify it

Reply:
Taking more insight on `GCLayout`, it's converting two representations back and forth. I'm not fully understood about GC info encoding and not sure whether it should be converted further.